src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+    * @param tolerance Tolerance for block volume calculations
+    * @param boundingBox Bounding box for input block
+    * @param origin Origin for input block
+    * @param desiredVolume Desired volume of lower left block cut by generated joint
+    * @param iterations Number of iterations that have been completed. Used to check that
+    *                   maximum number of iterations have not been exceeded
+    */
+  @tailrec
+  private def secantBisectionSolver(initialDist0: Double, initialDist1: Double, block: Block, 
+                                    tolerance: Double,
+                                    boundingBox: (Double, Double, Double, Double, Double, Double),
+                                    origin: (Double, Double, Double), desiredVolume: Double,
+                                    iterations: Int):
+                                    (Double, Double, Double) = {
+    val iterationLimit = 100
+    // Calculate diagonal length
+    val x_diff = boundingBox._4 - boundingBox._1
+    val y_diff = boundingBox._5 - boundingBox._2
+    val z_diff = boundingBox._6 - boundingBox._3
+    val diagonalLength = math.sqrt(x_diff*x_diff + y_diff*y_diff + z_diff*z_diff)
+
+    // Normal vector for processor joints
+    val normal = breeze.linalg.normalize(DenseVector[Double](boundingBox._4 - boundingBox._1,
+                                                             boundingBox._5 - boundingBox._2,
+                                                             boundingBox._6 - boundingBox._3))
+    val lowerLeft = DenseVector[Double](boundingBox._1, boundingBox._2, boundingBox._3)
+    // Centers for initial joint guesses
+    val center0Vec = lowerLeft + initialDist0*normal
+    val center0 = (center0Vec(0), center0Vec(1), center0Vec(2))
+    val center1Vec = lowerLeft + initialDist1*normal
+    val center1 = (center1Vec(0), center1Vec(1), center1Vec(2))
+
+    // Joint guesses and corresponding blocks
+    val joint0 = Joint((normal(0), normal(1), normal(2)), origin, center0, phi = 0.0, cohesion = 0.0,
+                       shape = Nil, processorJoint = true)
+    val joint1 = Joint((normal(0), normal(1), normal(2)), origin, center1, phi = 0.0, cohesion = 0.0,
+                       shape = Nil, processorJoint = true)
+    val blocks_0 = block.cut(joint0)
+    val blocks_1 = block.cut(joint1)
+    val blocks0 = blocks_0.map { case block @ Block(center, _) =>
+      Block(center, block.nonRedundantFaces)
+    }
+    val blocks1 = blocks_1.map { case block @ Block(center, _) =>
+      Block(center, block.nonRedundantFaces)
+    }
+
+    val sortedBlocks0 = blocks0.sortWith{ (left, right) =>
+      centroidCompare(left.centroid, right.centroid) }
+    val sortedBlocks1 = blocks1.sortWith{ (left, right) =>
+      centroidCompare(left.centroid, right.centroid) }
+
+    // Secant step
+    val a = Seq(initialDist0, initialDist1).min
+    val b = Seq(initialDist0, initialDist1).max
+    val f_a = if (initialDist1 > initialDist0) {
```

Comment:
You could do something similar to my comment above to set `f_a` and `f_b` at the same time.